WMF Framework/EventLoggingService.swift

```diff
@@ -596,3 +471,17 @@ public class EventLoggingService : NSObject, URLSessionDelegate {
         appInstallDate = nil
     }
 }
+
+extension EventLoggingService: PeriodicWorker {
+    public func doPeriodicWork(_ completion: @escaping () -> Void) {
+        tryPostEvents(completion)
+    }
+}
+
+extension EventLoggingService: BackgroundFetcher {
+    public func performBackgroundFetch(_ completion: @escaping (UIBackgroundFetchResult) -> Void) {
+        doPeriodicWork {
+            completion(.noData)
```

Comment:
I might not be understanding this right 🤔 but does this mean that `EventLoggingService` fetcher periodically tells the system that it didn't do anything (aka fetched no data) in the background? If a controller does nothing in the background, do we still want to make it a `BackgroundFetcher`?

Reply:
It's sending up events (via it's own `doPeriodicWork` method), but reporting `.noData` because technically nothing was downloaded. We could report `.newData` if events were successfully sent, though maybe we shouldn't make this a background task at all if it's uploading instead of downloading. https://developer.apple.com/library/archive/documentation/iPhone/Conceptual/iPhoneOSProgrammingGuide/BackgroundExecution/BackgroundExecution.html#//apple_ref/doc/uid/TP40007072-CH4-SW56